File: tools/commonist/src/scala/commonist/ui/NotWidePanel.scala
package commonist.ui

import java.awt.Dimension

import javax.swing.JPanel

// TODO make this a trait
class NotWidePanel(minimumWidth:Int) extends JPanel {
	override def getMinimumSize():Dimension = new Dimension(
			minimumWidth, 
			super.getMinimumSize.height)
}

File: tools/commonist-java/src/net/psammead/commonist/ui/NotWidePanel.java
package net.psammead.commonist.ui;

import java.awt.Dimension;

import javax.swing.JPanel;

public class NotWidePanel extends JPanel {
	private final int	minimumWidth;
	
	public NotWidePanel(int minimumWidth) {
		this.minimumWidth = minimumWidth;
	}
	
	@Override
	public Dimension getMinimumSize() {
		return new Dimension(
				minimumWidth, 
				super.getMinimumSize().height);
	}
}